File: VB/More/Config/ReadConfig/ReadConfig.vb
'------------------------------------------------------------------------------
' ReadConfig.vb
'
' Demonstrates how to read configuration settings.
'
' support@labjack.com
'------------------------------------------------------------------------------
Option Explicit On

Imports LabJack


Module ReadConfig

    Sub Main()
        Dim handle As Integer
        Dim numFrames As Integer
        Dim aNames() As String
        Dim aValues() As Double
        Dim errAddr As Integer
        Dim devType As Integer

        Try
            ' Open first found LabJack
            LJM.OpenS("ANY", "ANY", "ANY", handle)  ' Any device, Any connection, Any identifier
            'LJM.OpenS("T7", "ANY", "ANY", handle)  ' T7 device, Any connection, Any identifier
            'LJM.OpenS("T4", "ANY", "ANY", handle)  ' T4 device, Any connection, Any identifier
            'LJM.Open(LJM.CONSTANTS.dtANY, LJM.CONSTANTS.ctANY, "ANY", handle)  ' Any device, Any connection, Any identifier

            displayHandleInfo(handle)
            devType = getDeviceType(handle)

            ' Setup and call eReadNames to read configuration values from the
            ' LabJack.
            If devType = LJM.CONSTANTS.dtT4 Then
                ' LabJack T4 configuration to read
                numFrames = 8
                ReDim aNames(numFrames - 1)
                aNames(0) = "PRODUCT_ID"
                aNames(1) = "HARDWARE_VERSION"
                aNames(2) = "FIRMWARE_VERSION"
                aNames(3) = "BOOTLOADER_VERSION"
                aNames(4) = "SERIAL_NUMBER"
                aNames(5) = "POWER_ETHERNET_DEFAULT"
                aNames(6) = "POWER_AIN_DEFAULT"
                aNames(7) = "POWER_LED_DEFAULT"
            Else
                ' LabJack T7 and other devices configuration to read
                numFrames = 10
                ReDim aNames(numFrames - 1)
                aNames(0) = "PRODUCT_ID"
                aNames(1) = "HARDWARE_VERSION"
                aNames(2) = "FIRMWARE_VERSION"
                aNames(3) = "BOOTLOADER_VERSION"
                aNames(4) = "WIFI_VERSION"
                aNames(5) = "SERIAL_NUMBER"
                aNames(6) = "POWER_ETHERNET_DEFAULT"
                aNames(7) = "POWER_WIFI_DEFAULT"
                aNames(8) = "POWER_AIN_DEFAULT"
                aNames(9) = "POWER_LED_DEFAULT"
            End If
            ReDim aValues(numFrames - 1)
            LJM.eReadNames(handle, numFrames, aNames, aValues, errAddr)

            Console.WriteLine("")
            Console.WriteLine("Configuration settings:")
            For i = 0 To numFrames - 1
                Console.WriteLine("    " & aNames(i) & " : " & aValues(i).ToString("0.####"))
            Next
        Catch ljme As LJM.LJMException
            showErrorMessage(ljme)
        End Try

        LJM.CloseAll()  ' Close all handles

        Console.WriteLine("")
        Console.WriteLine("Done.")
        Console.WriteLine("Press the enter key to exit.")
        Console.ReadLine()  ' Pause for user
    End Sub

End Module

File: C#/More/Config/ReadConfig/ReadConfig.cs
//-----------------------------------------------------------------------------
// ReadConfig.cs
//
// Demonstrates how to read configuration settings on a LabJack.
//
// support@labjack.com
//
// Relevant Documentation:
//
// LJM Library:
//     LJM Library Installer:
//         https://labjack.com/support/software/installers/ljm
//     LJM Users Guide:
//         https://labjack.com/support/software/api/ljm
//     Opening and Closing:
//         https://labjack.com/support/software/api/ljm/function-reference/opening-and-closing
//     eReadNames:
//         https://labjack.com/support/software/api/ljm/function-reference/ljmereadnames
//
// T-Series and I/O:
//     Modbus Map:
//         https://labjack.com/support/software/api/modbus/modbus-map
//     Hardware Overview (Device Information Registers):
//         https://labjack.com/support/datasheets/t-series/hardware-overview
//-----------------------------------------------------------------------------

using System;
using LabJack;


namespace ReadConfig
{
    class ReadConfig
    {
        static void Main(string[] args)
        {
            ReadConfig rc = new ReadConfig();
            rc.performActions();
        }

        public void showErrorMessage(LJM.LJMException e)
        {
            Console.Out.WriteLine("LJMException: " + e.ToString());
            Console.Out.WriteLine(e.StackTrace);
        }

        public void performActions()
        {
            int handle = 0;
            int devType = 0;
            int conType = 0;
            int serNum = 0;
            int ipAddr = 0;
            int port = 0;
            int maxBytesPerMB = 0;
            string ipAddrStr = "";
            string[] aNames;
            double[] aValues;
            int numFrames = 0;
            int errAddr = 0;

            try
            {
                //Open first found LabJack
                LJM.OpenS("ANY", "ANY", "ANY", ref handle);
                //devType = LJM.CONSTANTS.dtANY; //Any device type
                //conType = LJM.CONSTANTS.ctANY; //Any connection type
                //LJM.Open(devType, conType, "ANY", ref handle);

                LJM.GetHandleInfo(handle, ref devType, ref conType, ref serNum, ref ipAddr, ref port, ref maxBytesPerMB);
                LJM.NumberToIP(ipAddr, ref ipAddrStr);
                Console.WriteLine("Opened a LabJack with Device type: " + devType + ", Connection type: " + conType + ",");
                Console.WriteLine("Serial number: " + serNum + ", IP address: " + ipAddrStr + ", Port: " + port + ",");
                Console.WriteLine("Max bytes per MB: " + maxBytesPerMB);

                //Setup and call eReadNames to read config. values from the
                //LabJack.
                if (devType == LJM.CONSTANTS.dtT4)
                {
                    //LabJack T4 configuration to read
                    aNames = new string[] {
                        "PRODUCT_ID", "HARDWARE_VERSION",
                        "FIRMWARE_VERSION", "BOOTLOADER_VERSION",
                        "SERIAL_NUMBER", "POWER_ETHERNET_DEFAULT",
                        "POWER_AIN_DEFAULT", "POWER_LED_DEFAULT"
                    };
                }
                else
                {
                    //LabJack T7 and other devices configuration to read
                    aNames = new string[] {
                        "PRODUCT_ID", "HARDWARE_VERSION",
                        "FIRMWARE_VERSION", "BOOTLOADER_VERSION",
                        "WIFI_VERSION", "SERIAL_NUMBER",
                        "POWER_ETHERNET_DEFAULT", "POWER_WIFI_DEFAULT",
                        "POWER_AIN_DEFAULT", "POWER_LED_DEFAULT"
                    };
                }
                numFrames = aNames.Length;
                aValues = new double[numFrames];
                errAddr = 0;
                LJM.eReadNames(handle, numFrames, aNames, aValues, ref errAddr);

                Console.WriteLine("\nConfiguration settings:");
                for (int i = 0; i < numFrames; i++)
                    Console.WriteLine("    " + aNames[i] + " : " + aValues[i].ToString("0.####"));
            }
            catch (LJM.LJMException e)
            {
                showErrorMessage(e);
            }

            LJM.CloseAll();  //Close all handles

            Console.WriteLine("\nDone.\nPress the enter key to exit.");
            Console.ReadLine();  // Pause for user
        }
    }
}
